Google APIs: point to Flutter docs and Dart sample (#3238)
dart-lang/site-www: src/_guides/google-apis.md

such as the Google Docs API, YouTube Data API,
Cloud Translation API, and Cloud Storage API.

> If you're building a Flutter application, see the corresponding
> [Google APIs guide][flutter-google-apis].

Reviewer: ```suggestion
If you're building a Flutter application, see the
[Flutter guide for Google APIs][flutter-google-apis].
```

> If you'd like to access Google APIs as part of a server application,
> [see this sample][server-sample].

Reviewer: ```suggestion
If you'd like to use Google APIs as part of a server application, see the
[google_apis server sample][server-sample].
```
